src/gf.c

```diff
@@ -2237,7 +2237,23 @@ static jl_tupletype_t *lookup_arg_type_tuple(jl_value_t *arg1 JL_PROPAGATES_ROOT
     return jl_lookup_arg_tuple_type(arg1, args, nargs, 1);
 }
 
-jl_method_instance_t *jl_method_lookup(jl_value_t **args, size_t nargs, size_t world)
+JL_DLLEXPORT jl_method_instance_t *jl_method_lookup_by_tt(jl_tupletype_t *tt, size_t world)
+{
+    // TODO: The fast path is heavily optimized for args and avoids forming the signature type
+    jl_methtable_t *mt = jl_gf_ft_mtable(jl_tparam0(tt));
```

Comment:
Make this an argument too, ref https://github.com/JuliaGPU/GPUCompiler.jl/issues/530#issuecomment-1808020702, probably addressing https://github.com/JuliaGPU/GPUCompiler.jl/issues/208?

Reply:
That would require an extra ccall to `jl_method_table_for`

Reply:
Not if optional, which I see is what you added.